src/components/AddressForm.tsx

```diff
@@ -1,121 +1,150 @@
 import Navigation from '@libs/Navigation/Navigation';
 import * as ValidationUtils from '@libs/ValidationUtils';
 import CONST from '@src/CONST';
+import type {AddressForm as AddressFormValues} from '@src/types/onyx';
 import AddressSearch from './AddressSearch';
 import CountrySelector from './CountrySelector';
 import FormProvider from './Form/FormProvider';
 import InputWrapper from './Form/InputWrapper';
 import StatePicker from './StatePicker';
 import TextInput from './TextInput';
 
-const propTypes = {
+type AddressFormProps = {
     /** Address city field */
-    city: PropTypes.string,
+    city?: string;
 
     /** Address country field */
-    country: PropTypes.string,
+    country?: string;
 
     /** Address state field */
-    state: PropTypes.string,
+    state?: string;
 
     /** Address street line 1 field */
-    street1: PropTypes.string,
+    street1?: string;
 
     /** Address street line 2 field */
-    street2: PropTypes.string,
+    street2?: string;
 
     /** Address zip code field */
-    zip: PropTypes.string,
+    zip?: string;
 
     /** Callback which is executed when the user changes address, city or state */
-    onAddressChanged: PropTypes.func,
+    onAddressChanged?: OnAddressChanged;
 
     /** Callback which is executed when the user submits his address changes */
-    onSubmit: PropTypes.func.isRequired,
+    onSubmit: () => void;
 
     /** Whether or not should the form data should be saved as draft */
-    shouldSaveDraft: PropTypes.bool,
+    shouldSaveDraft?: boolean;
 
     /** Text displayed on the bottom submit button */
-    submitButtonText: PropTypes.string,
+    submitButtonText?: string;
 
     /** A unique Onyx key identifying the form */
-    formID: PropTypes.string.isRequired,
+    formID: string;
 };
 
-const defaultProps = {
-    city: '',
-    country: '',
-    onAddressChanged: () => {},
-    shouldSaveDraft: false,
-    state: '',
-    street1: '',
-    street2: '',
-    submitButtonText: '',
-    zip: '',
-};
+type OnAddressChanged = (data: string, key: string) => void;
 
-function AddressForm({city, country, formID, onAddressChanged, onSubmit, shouldSaveDraft, state, street1, street2, submitButtonText, zip}) {
+function AddressForm({
+    city = '',
+    country = '',
+    formID,
+    onAddressChanged = () => {},
+    onSubmit,
+    shouldSaveDraft = false,
+    state = '',
+    street1 = '',
+    street2 = '',
+    submitButtonText = '',
+    zip = '',
+}: AddressFormProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
-    const zipSampleFormat = lodashGet(CONST.COUNTRY_ZIP_REGEX_DATA, [country, 'samples'], '');
+
+    let zipSampleFormat = '';
+
+    if (country && country in CONST.COUNTRY_ZIP_REGEX_DATA) {
+        const countryData = CONST.COUNTRY_ZIP_REGEX_DATA[country as keyof typeof CONST.COUNTRY_ZIP_REGEX_DATA];
+        if (countryData && 'samples' in countryData) {
+            zipSampleFormat = countryData.samples;
+        }
+    }
```

Comment:
Could you check in what format `country` can be passed to this component?  Maybe we can simplify this part if type country as
`country?: keyof typeof CONST.COUNTRY_ZIP_REGEX_DATA;`
or
` country?: keyof typeof CONST.COUNTRY;`

Reply:
We can't do it here in such a way, since we  can have an empty string as a value here (default value or changed)

Reply:
Can we remove the default value and use nullish coaleascing where necessary?
If for some reason it's not possible I think we can add `| '' `to the type

Reply:
No, it is not posible

1. String can be still passed from the upper level  
2. If we add `| ''` - we still need this check. It will be highlighted that it is possibly string

Reply:
But with using ` | ''` you can get rid of part of checks
If `country?: keyof typeof CONST.COUNTRY_ZIP_REGEX_DATA | '';`
```
 if (country) {
        const countryData = CONST.COUNTRY_ZIP_REGEX_DATA[country];
        if (countryData && 'samples' in countryData) {
            zipSampleFormat = countryData.samples;
        }
    }
```

And if it's possible to type  `country?: keyof typeof CONST.COUNTRY | '';` - you need to check what value is passed as a country, then it can be simplified to
```
const zipSampleFormat = country ? CONST.COUNTRY_ZIP_REGEX_DATA[country]?.samples : '';
```
What do you think?